### DISCUSSION TITLE
Is there a way to make bar/panel stay still when switching workspaces using vswitch?

### DISCUSSION THREAD

**Discussion by A7R7** [WayfireWM/wayfire#2066] [Q&A] [answered]
I tried to set wf-panel and waybar's layer to top/bottom/overlay/background, all faied to make the panel stay still when switching the workspaces. I suspect I may have to edit the source code to achieve this.

**Comment by ammen99** [comment]
There is currently no such option, it will need tweaking of the vswitch plugin.

**Reply by A7R7** [reply]
I took a brief look into vswitch's code but failed to understand how it worked (not very good at cpp :( ,  I may spend more time on these). But guessing from the code comment, vswitch is probably drawing an overlay during switching of workspaces. Is that right?

Also, some other plugins may use vswitch to work. Will it break them if I tweak vswitch?

**Reply by ammen99** [reply, marked as answer]
I dont think such a change will break anything.

If you are looking at this in the code: vswitch uses workspace-wall.hpp for rendering. The wall has a render node added to the scene. To make the panel stay on top, you'd need the following changes:

- In the workspace nodes (their implementation is in core, but they are used by workspace-wall) do not generate instances for overlay and top layers.
- Add the wall render node below those layers in the scenegraph
